packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaForm.tsx

```diff
@@ -238,6 +238,12 @@ function TableColumnSchemaFormInternal(
           return !currentFormData.find(fd => fd.name === cm.name)
         },
       )
+      // Delete the ID column so TableColumnSchemaUtils.createTableUpdateTransactionRequest recognizes these as new columns
+      // createTableUpdateTransactionRequest uses the existing column ID to track column updates in the table, so any new columns should have no ID
+      // The user can also modify these columns before submitting them, so the ID may not be accurate when we end up submitting them
+      columnsToAdd.forEach(cm => {
+        delete cm.id
+      })
```

Comment:
This is inside of the function called when _existing_ columns are added to a table. Examples include 
* Columns imported from another table
* Columns for a view that correspond to annotations on objects in the view

The update utility expects that columns that are _new to the table_ do not include IDs, so we remove them here. This is how the Java implementation works. 

To me, this feels a bit dangerous. I personally didn't quite grasp how this worked until I tried to fix a few bugs here. I think it might feel less 'dangerous' if we were to track our own set of changes instead of relying on presence/absence of IDs, but that is a pretty large change since it would also affect the editor implementation.